src/main/scala/it/pagopa/interop/purposeprocess/api/impl/PurposeVersionActivation.scala

```diff
@@ -68,10 +45,25 @@ final case class PurposeVersionActivation(
     ownership: Ownership
   )(implicit contexts: Seq[(String, String)]): Future[PurposeVersion] = {
 
-    def waitForApproval(changedBy: ChangedBy): Future[PurposeVersion] =
+    def changeToWaitForApproval(changedBy: ChangedBy): Future[PurposeVersion] =
       purposeManagementService.waitForApprovalPurposeVersion(purpose.id, version.id, StateChangeDetails(changedBy))
 
-    def activate(changedBy: ChangedBy): Future[PurposeVersion] = {
+    def createWaitForApproval(): Future[PurposeVersion] = for {
+      _                         <- Future.traverse(purpose.versions.find(_.state == WAITING_FOR_APPROVAL).toList) { v =>
+        purposeManagementService.deletePurposeVersion(purpose.id, v.id)
+      }
+      draftVersion              <- purposeManagementService.createPurposeVersion(
+        purpose.id,
+        PurposeVersionSeed(version.dailyCalls, version.riskAnalysis)
+      )
+      waitingForApprovalVersion <- purposeManagementService.waitForApprovalPurposeVersion(
```

Comment:
This double step is not a very clean solution, but right now it is not possible to directly create a waiting for approval version because the business logic and the checks are performed on the management

Reply:
Until we have the logic inside management, I don't see a different solution from what you have implemented.
Just a minor suggestion: I would pass the version explicitly in functions `changeToWaitForApproval` and `createWaitForApproval` like we do in `activate`.